__tests__/services/FunctionProcessor.test.ts

```diff
@@ -0,0 +1,201 @@
+import { processFunction } from "../../src/services/BlockService/FunctionProcessor";
+import { Block } from "../../src/types/Block";
+import { Params } from "../../src/types/Params";
+
+describe('processFunction', () => {
+    it('should return undefined if block has no function', async () => {
+        const block: Block = { function: undefined };
+        const params: Params = {
+            userInput: "",
+            currPath: null,
+            prevPath: null,
+            goToPath: function (pathToGo: string): void {
+                throw new Error("Function not implemented.");
+            },
+            setTextAreaValue: function (value: string): void {
+                throw new Error("Function not implemented.");
+            },
+            injectMessage: function (content: string | JSX.Element, sender?: string | undefined): Promise<string | null> {
+                throw new Error("Function not implemented.");
+            },
+            streamMessage: function (content: string | JSX.Element, sender?: string | undefined): Promise<string | null> {
+                throw new Error("Function not implemented.");
+            },
+            removeMessage: function (id: string): Promise<string | null> {
+                throw new Error("Function not implemented.");
+            },
+            endStreamMessage: function (sender: string): Promise<boolean> {
+                throw new Error("Function not implemented.");
+            },
+            showToast: function (content: string | JSX.Element, timeout?: number | undefined): void {
+                throw new Error("Function not implemented.");
+            },
+            dismissToast: function (id: string): string | null {
+                throw new Error("Function not implemented.");
+            },
+            openChat: function (isOpen: boolean): void {
+                throw new Error("Function not implemented.");
+            }
+        };
```

Comment:
This is a little too verbose, consider mocking with `jest.fn()` instead and defining it for each test case using `beforeEach` - can reference other test cases 😄

Reply:
Thank you @tjtanjin , I will update my PR soon